### PR TITLE
Add custom action to freeze a frame using a MUX node

### PR DESCRIPTION
**Pull request by fredwarren**
This script adds a custom action to the Batch node. When used, a MUX node is connected to the selected node and freezes the current frame.

### CODE REVIEW

File: custom-actions/batch/freeze_frame_using_mux.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        for context in range(1, 11):
+            if contexts.has_key(context) is False:
+                return context
+        return 0
+
+    def scope_node(selection):
+        """Scope the custom action to the nodes residing in a Batch Schematic."""
+        import flame
+        for item in selection:
+            if isinstance(item, (flame.PyNode)):
```
Comment (jpbrault) [inline custom-actions/batch/freeze_frame_using_mux.py]: Could remove extra () pairs

File: custom-actions/batch/freeze_frame_using_mux.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    """Return custom actions to execute on Batch objects."""
+
+    def get_context():
+        """Get the first available slot in a Batch Group's Context views list."""
+        import flame
+        contexts = flame.batch.contexts
+        for context in range(1, 11):
+            if contexts.has_key(context) is False:
+                return context
+        return 0
```
Comment (jpbrault) [inline custom-actions/batch/freeze_frame_using_mux.py]: If 0 is not a valid valid, you could return None instead to make it explicit that this is not a number.

File: custom-actions/batch/freeze_frame_using_mux.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
+"""Connect a MUX to a node and freeze the current frame"""
+
+def get_batch_custom_ui_actions():
+    """Return custom actions to execute on Batch objects."""
+
+    def get_context():
+        """Get the first available slot in a Batch Group's Context views list."""
+        import flame
+        contexts = flame.batch.contexts
+        for context in range(1, 11):
```
Comment (jpbrault) [inline custom-actions/batch/freeze_frame_using_mux.py]: Might be worth having constant for 1 and 11.

**Review by jpbrault — DISMISSED** [review]

File: custom-actions/batch/freeze_frame_using_mux.py (hunk 3 above)
Comment (fredwarren) [reply custom-actions/batch/freeze_frame_using_mux.py]: Should I put them at the beginning of the file, the beginning of the get_ context function or above the for loop?

**Review by jpbrault — APPROVED** [review]

**Pull request merged**